### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by emirotin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,218 +48,6 @@ var token = require('resin-token')({
...
-**Kind**: static method of <code>[token](#module_token)</code>  
-**Summary**: Check if the given token has expired  
-**Returns**: {Promise<boolean>  
-**Access:** public  
-**Example**  
-```js
-token.isExpired(jwtToken).then (isExpired) ->
-		console.log(isExpired)
-```
-
```
Comment (pimterry) [inline README.md]: This removes all the docs, we should definitely keep those.

File: lib/token.ts [hunk 2]
```diff
@@ -0,0 +1,330 @@
+/*
+ * decaffeinate suggestions:
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS201: Simplify complex destructure assignments
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
```
Comment (pimterry) [inline lib/token.ts]: We can drop this

File: lib/token.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+		return this.storage.remove(TOKEN_KEY);
+	}
+
+	/**
+	 * @summary Get the saved token data
+	 * @function
+	 * @public
+	 *
+	 * @description
+	 * In this context, "data" refers to the information encoded in the token.
```
Comment (pimterry) [inline lib/token.ts]: I think it'd be useful to note that this resolves to undefined if there is no token available.

File: lib/token.ts [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+	}
+
+	/**
+	 * @summary Get a property from a saved token
+	 * @function
+	 * @public
+	 *
+	 * @description
+	 * This function resolves to undefined for any property name if there is no token.
+	 * It also resolved to undefined if the property name is invalid.
```
Comment (pimterry) [inline lib/token.ts]: Also resolves to undefined if there is no token available.

File: lib/token.ts [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+	 *
+	 * @description
+	 * This function does't save the token. Use `token.set()` if you want to persist it afterwards.
+	 *
+	 * @param {String} token - token
+	 * @returns {Promise<Object>} parsed token
+	 *
+	 * @example
+	 * token.parse('...').then (parsedToken) ->
+	 * 	console.log(parsedToken)
```
Comment (pimterry) [inline lib/token.ts]: All these code examples are still coffeescript

**Review by pimterry — CHANGES_REQUESTED** [review]

**Comment by pimterry** [comment]
Bonus point: now that we're sure this `@resin-io/types-*` approach is actually practical, we should raise it in a process call to check everybody else is happy with it, add it to the TypeScript guidelines, and bring it up on a monday call, so everybody else starts doing it too.

File: lib/token.ts (hunk 2 above)
Comment (emirotin) [reply lib/token.ts]: Missed in this file

File: README.md (hunk 1 above)
Comment (emirotin) [reply README.md]: Hmm, of course we should, I wonder what has happened.

Comment (emirotin) [reply README.md]: Probably related to conversion to a class, will have to tinker with JSDoc a bit

**Comment by emirotin** [comment]
@pimterry do you remember if we had AppVeyor for this repo? We don't have it now at least

File: lib/token.ts (hunk 4 above)
Comment (emirotin) [reply lib/token.ts]: Actually it's said in the first line

**Comment by emirotin** [comment]
@pimterry please re-review

**Review by pimterry — APPROVED** [review]
Yep, needs squashing too, but otherwise looks great to me. Sorry, I have no idea if we used to have appveyor...

**Comment by emirotin** [comment]
I'm pretty sure we had because we have the config file for it. I've asked Apo to switch it back on

**Pull request merged**